tests/python/pants_test/source/test_source_root.py

```diff
@@ -91,6 +91,51 @@ def root(path, langs):
     self.assertEqual(root('src/go/src', ('go',)),
                       trie.find('src/go/src/foo/bar/baz.go'))
 
+  def test_source_root_trie_traverse(self):
```

Comment:
Return type is `-> None`. The benefit of doing that is that MyPy will then know to check this code + it's easier to scan what the function does.

Reply:
Should we be putting `-> None` type hints in test functions that have no meaningful return values?

Reply:
Yes, encouraged to do this so that MyPy knows to type check the code and to catch the unlikely but possible bug where we try returning from a unit test. 

As close as we can get to 100%, the better. Makes it simpler to always aim for 100% and have less special cases.